config/element-api.php

```diff
@@ -336,7 +404,15 @@ function getListing($locale)
         'criteria' => $searchCriteria,
         'transformer' => function (Entry $entry) use ($locale, $pagePath) {
 
+            $status = 'live';
+            $revision = getDraftOrVersionOfEntry($entry);
+            if ($revision) {
+                $entry = $revision['entry'];
+                $status = $revision['status'];
+            }
+
```

Comment:
This whole block feels a bit ugly to have to duplicate each time we want to support drafts... any thoughts on how we could make it a bit nicer?

Reply:
Maybe move the logic up into the `getDraftOrVersionOfEntry` function, i.e. that should return an entry with the status on it otherwise, return the plain entry that was passed in with the `'live` status?

``` php
$entry = getDraftOrVersionOfEntry($entry);
$entry->status // draft
```